packages/astro/src/assets/types.ts

```diff
@@ -151,6 +152,12 @@ type ImageSharedProps<T> = T & {
 	 * ```
 	 */
 	quality?: ImageQuality;
+
+	layout?: ImageLayout;
+
+	fit?: 'fill' | 'contain' | 'cover' | 'none' | 'scale-down' | (string & {});
+
+	position?: string;
```

Comment:
I've gone with the shorter names for the HTML attributes for ease of use and avoidance of confusion of camel vs kebab